src/interface_matlab.cpp

```diff
@@ -158,6 +158,15 @@ void amici_daxpy(int n, double alpha, const double *x, const int incx, double *y
     FORTRAN_WRAPPER(daxpy)(&n_, &alpha, x, &incx_, y, &incy_);
 }
 
+/** conversion from mxArray to vector<realtype>
+ *  @param x input
+ *  @return int_x casted value
+ */
+std::vector<realtype> mxArrayToVector(mxArray *mexData, int dataLength){
+    std::vector<realtype> dataVector(dataLength);
```

Comment:
I'm fine with this, but @dweindl is a strong proponent of `auto`. Could you replace all `std::vector<realtype>` initializations with `auto`?

Reply:
also you might want to rebase as everything before d0cd3b7 is still from another PR.

Reply:
I implemented the auto where it was possible and would do a squash commit anyway... In that case: also rebase?